src/roles/succubus.py

```diff
@@ -0,0 +1,315 @@
+        if votee is None:
+            ENTRANCED_DYING.clear()
+            return
+        ENTRANCED_DYING.difference_update(not_lynching)
+
+@event_listener("chk_decision_lynch", priority=5)
+def on_chk_decision_lynch(evt, cli, var, voters):
+    # a different event may override the original votee, but people voting along with succubus
+    # won't necessarily know that, so base whether or not they risk death on the person originally voted
+    _kill_entranced_voters(var, evt.params.votelist, evt.params.not_lynching, evt.params.original_votee)
+
+@event_listener("chk_decision_abstain")
+def on_chk_decision_abstain(evt, cli, var, not_lynching):
+    _kill_entranced_voters(var, evt.params.votelist, not_lynching, None)
+
+# entranced logic should run after team wins have already been determined (aka run last)
+# we do not want to override the win conditions for neutral roles should they win while entranced
+# For example, entranced monsters should win with other monsters should mosnters win, and be
+# properly credited with a team win in that event.
+@event_listener("player_win", priority=6)
+def on_player_win(evt, var, user, role, winner, survived):
+    nick = user.nick
+    if nick in ENTRANCED:
+        evt.data["special"].append("entranced")
+        if winner != "succubi" and role not in var.TRUE_NEUTRAL_ROLES:
+            evt.data["won"] = False
+        else:
+            evt.data["iwon"] = True
+    if role == "succubus" and winner == "succubi":
+        evt.data["won"] = True
+
+@event_listener("chk_win", priority=2)
+def on_chk_win(evt, cli, var, rolemap, lpl, lwolves, lrealwolves):
+    lsuccubi = len(rolemap.get("succubus", ()))
+    lentranced = len(ENTRANCED - var.DEAD)
+    if var.PHASE == "day" and lpl - lsuccubi == lentranced:
+        evt.data["winner"] = "succubi"
+        evt.data["message"] = messages["succubus_win"].format(plural("succubus", lsuccubi), plural("has", lsuccubi), plural("master's", lsuccubi))
+
+@event_listener("del_player")
+def on_del_player(evt, cli, var, nick, nickrole, nicktpls, death_triggers):
+    if nickrole != "succubus":
+        return
+    if nick in VISITED:
+        # if it's night, also unentrance the person they visited
+        if var.PHASE == "night" and var.GAMEPHASE == "night":
+            if VISITED[nick] in ENTRANCED:
+                ENTRANCED.discard(visited[nick])
+                ENTRANCED_DYING.discard(visited[nick])
+                pm(cli, VISITED[nick], messages["entranced_revert_win"])
+        del VISITED[nick]
+
+    # if all succubi are dead, one of two things happen:
+    # 1. if all succubi idled out (every last one of them), un-entrance people
+    # 2. otherwise, kill all entranced people immediately, they still remain entranced (and therefore lose)
+    # death_triggers is False for an idle-out, so we use that to determine which it is
+    if death_triggers:
+        ALL_SUCC_IDLE = False
+    if len(var.ROLES["succubus"]) == 0:
+        if ALL_SUCC_IDLE:
```

Comment:
This won't work - `ALL_SUCC_IDLE` has been assigned as a local variable, but if it hasn't been set to anything, it _will not_ fall back to the global one. Choices:

- Use a differently-named local variable, which you originally assign to the global version, then update that (If you don't want the global one to change)
- Add a `global` statement for the variable at the top of the function (If you wish to change the global variable to change)
- Use a mutable data structure, like a list, which you can then do stuff with to avoid the `global` shenanigans